tests/phpunit/tests/filesystem/direct/chgrp.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * Tests for the WP_Filesystem_Direct::chgrp() method.
+ *
+ * @package WordPress
+ */
+
+require_once __DIR__ . '/base.php';
+
+/**
+ * @group admin
+ * @group filesystem
+ * @group filesystem-direct
+ *
+ * @covers WP_Filesystem_Direct::chgrp
+ */
+class Tests_Filesystem_WpFilesystemDirect_Chgrp extends WP_Filesystem_Direct_UnitTestCase {
```

Comment:
Further testing requires more investigation for stable testing. This includes a source change to support detecting `root` as the user to prevent test failures in local test runners for methods such as `::owner()` and `::group()`. The results of further investigation, and the above source change will be proposed in separate tickets/PRs.

🔢 Applies elsewhere in the tests / `WP_Filesystem_Direct` class.